routing/base/astar_algorithm.hpp

```diff
@@ -349,6 +369,14 @@ typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPathBidirect
         if (!foundAnyPath || bestPathReducedLength > curPathReducedLength)
         {
           bestPathReducedLength = curPathReducedLength;
+          bestPathRealLength = stateV.distance + len - pV + distW;
+          bestPathRealLength += cur->pS;
+
+          if (distW > 0)
```

Comment:
Why do we need this condition? For vertices on the shortest path reduced length can be zero, but real length is not zero (imagine a straight line between two points on a plane, and each edge is just a segment on the line. Reduced length of each edge is zero, if we assume Euclidian metric, but real length can be non-zero).

Also, for symmetry and readability, why not to rewrite formula in the following way:
```c++
bestPathRealLength = stateV.distance + len + distW;
bestPathRealLength += cur->pS - pV;
bestPathRealLength += nxt->pS - next->ConsistentHeuristic(stateW.vertex);
```
?